base/multidimensional.jl

```diff
@@ -479,6 +479,53 @@ end
     end
 end
 
+# accumulate_pairwise is on the order of 50% slower then accumulate, but more numerically
+# stable in certain situtations (e.g. sums)
+function _accumulate_pairwise!{T}(op, c::AbstractVector{T}, v::AbstractVector, s, i1, n)
+    local s_::T # for sum(v[range(i1,n)]), i.e. sum without s
+    @inbounds if n < 128
+        s_ = v[i1]
+        c[i1] = op(s, s_)
+        for i = i1+1:i1+n-1
+            s_ = op(s_, v[i])
+            c[i] = op(s, s_)
+        end
+    else
+        n2 = n >> 1
+        s_ = _accumulate_pairwise!(op, c, v, s, i1, n2)
+        s_ = op(s_, _accumulate_pairwise!(op, c, v, op(s, s_), i1+n2, n-n2))
+    end
+    return s_
+end
+
+function accumulate_pairwise!(op, result::AbstractVector, v::AbstractVector)
```

Comment:
Does this need `function accumulate_pairwise!{F}(op::F, result::AbstractVector, v::AbstractVector)` Ref: https://github.com/JuliaLang/julia/issues/18207